scripts/src/pullrequest/metadata.py

```diff
@@ -0,0 +1,155 @@
+
+    Notably, this functionality does NOT pull version information for any submission.
+
+    Args:
+        pr_api_url (str): URL of the GitHub PR
+
+    Raises:
+        NoMatchesError: In cases where the PR did not contain any files that
+          matched the expected format for a chart contribution.
+        MultipleMatchesError: In cases where the PR contained changes to multiple
+          charts, which is explicitly forbidden.
+
+    Returns:
+        Returns the category, organization, and name. E.g. "partner",
+        "hashicorp", "vault".
+    """
+
+    modified_files = prartifact.get_modified_files(pr_api_url)
+    first_match = None
+    modification_matcher = re.compile(
+        matchers.submission_path_matcher(
+            strict_categories=True, include_version_matcher=False
+        )
+    )
+    for file in modified_files:
+        match = modification_matcher.match(file)
+        if not match:
+            continue
+
+        # Store the value of our groupings the first time we match
+        # so that we can track if it changes.
+        if not first_match:
+            first_match = match
+            continue
+
+        if match.groups() != first_match.groups():
+            raise MultipleMatchesError(
+                first_match=first_match.groups(),
+                subsequent_match=match.groups(),
+            )
+
+    # We didn't find a match, so we need to bail.
+    if not first_match:
+        raise NoMatchesError
+
+    cat, org, name = first_match.groups()
+    # normalize the partners directory to partner
+    cat = "partner" if cat == "partners" else cat
+    return cat, org, name
+
+
+def main():
+    """CLI for determining the the cat/org/chart being modified.
+
+    Return codes:
+        0:  No problems.
+        10: Multiple matches were found.
+        20: No matches were found.
+    """
+    parser = ArgumentParser(
+        description="""Read the modified files from a GitHub Pull Requests
+and determine the category, organization, and chartname being modified.
+
+This information is extracted from the relative path of the files being
+modified. Specifically, this reviews the charts/ directory and looks for the
+subsequent path to determine the category, organization, and chart name.
+"""
+    )
+    parser.add_argument(
+        "api_url",
+        help="The API URL for a given pull request. E.g. https://api.github.com/repos/openshift-helm-charts/charts/pulls/1",
+        type=str,
+    )
+    parser.add_argument(
+        "-g",
+        "--emit-to-github-output",
+        help="Sends keys and values to the $GITHUB_OUTPUT.",
+        default=False,
+        dest="emit_to_github",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--ignore-error-no-matches",
+        help="When enabled, returns a zero exit code instead of a non-zero for cases where no matches were found",
+        dest="ignore_no_matches",
+        action="store_true",
+        default=False,
+    )
```

Comment:
Is this flag motivated by an actual use case ?

Reply:
I don't see a use case in it in my dev repo, and I don't remember why I included it off hand. If you'd prefer it, I can remove the flag.

Reply:
Yes, I would be in favor of removing this code if it's not needed.

Reply:
Fixed and pushed here: https://github.com/openshift-helm-charts/development/pull/314/commits/035b0274a32185316345cb8b7926bac5f35d35cd

Tests have been canceled and should be rerunning now.